src/fitnesse/wiki/WikiPage.java

```diff
@@ -79,6 +79,8 @@
 
   PageCrawler getPageCrawler();
 
+  PageCrawler getPageCrawler(PagePruningStrategy strategy);
```

Comment:
I think the interface (which is already quite scattered functionality-wise) should not be extended with such specific methods I think. Wouldn't it be nicer to create an adapter class that implements the pruning strategy on top of the original PageCrawler?

Reply:
I'm not sure I can follow you here.
Your point (not to extend the already broad interface with specific methods) is valid, and I'd like to follow through on this. But I don't really understand your suggestion; can you please elaborate a bit more?

Also, I'm wondering whether this is indeed a specific method -- isn't the ability to prune a search tree part of the "core business" of the crawler? Following this thought further, maybe it would be a better solution to remove the getPageCrawler() method without arguments and implement a NoPruning strategy, so that these two ways of handling the crawling can be unified here, in the WikiPageFinder and in the PageCrawler itself?
I'd appreciate your thoughts on this.

Reply:
> ...can you please elaborate a bit more?

My hunch is to do something like

    crawler = new PruningCrawler(wikiPage.getPageCrawler())

`PruningCrawler` would act as a filter, e.i. it would remove all pruned pages.

> Also, I'm wondering whether this is indeed a specific method -- isn't the ability to prune a search tree part of the "core business" of the crawler?

I suppose it is :). How about if the PageCrawler would have a method `traverse` to which you can provide a strategy? At least that would move the concern of traversing/pruning down into the page crawler and the wiki page itself is not bothered with it?

 Adding an additional argument to `WikiPage.getPageCrawler()` would only make the interface more complicated, so I would not be in favour of this.

Reply:
> PruningCrawler would act as a filter, e.i. it would remove all pruned pages.

The problem with this approach is that it is not efficient. When looking at any given page in isolation, how would I be able to figure out whether it needs to be removed? If the page itself is pruned, that's easy; but if it isn't, I have to follow the parent pages up to the root page to decide whether one of the parent pages is pruned. This gives us an exponential algorithm (in the worst case), I believe, and I'm not a big fan of this. That's why I tried to follow the other approach: not traverse the skipped page tree parts in the first place.

I will look into moving the strategy parameter from the getPageCrawler() method to the crawler's traversal. Makes more sense altogether, I guess. Thanks for the suggestion!